spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/TaskNode.java

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.core.dsl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+
+import org.springframework.cloud.dataflow.core.dsl.graph.Graph;
+import org.springframework.util.Assert;
+
+/**
+ * The root AST node for any AST parsed from a composed task specification.
+ *
+ * @author Andy Clement
+ */
+public class TaskNode extends AstNode {
+
+	/**
+	 * The name of the composed task.
+	 */
+	private String name;
+	
+	/**
+	 * The DSL text that was parsed to create this ComposedTaskNode.
+	 */
+	private String composedTaskText;
+
+	/**
+	 * The sequence of LabelledNodes parsed from the specification.
+	 */
+	private List<LabelledTaskNode> sequences;
+
+	/**
+	 * All the apps mentioned in the composed task definition.
+	 */
+	private List<TaskApp> taskApps;
+	
+	TaskNode(String name, String composedTaskSpecification, List<LabelledTaskNode> sequences, boolean inAppMode) {
+		super((sequences.size()==0) ? 0 : sequences.get(0).getStartPos(), 
+				(sequences.size()==0) ? 0 : sequences.get(sequences.size()-1).getEndPos());
+		this.name = name;
+		this.composedTaskText = composedTaskSpecification;
+		this.sequences = sequences;
+		// TODO use inAppMode to police what can be called on this node?
```

Comment:
If I do that I get an 'unused variable' warning on it since no-one ever does anything with it (at the moment), for now I just left it in the ctor args and the comment.